sources/org.osbuild.ostree

```diff
@@ -63,6 +65,40 @@ SCHEMA = """
 """
 
 
+class OstreeResult(SourceResult):
+
+    REMOTE_ADD = 1
+    GPG_IMPORT = 2
+    PULL = 3
+    REMOTE_DELETE = 4
+    DONE = 5
+
+    def __init__(self, source: Dict):
+        super().__init__(OstreeResult.format_source(source))
+        self.ostree_error = None
+        self.state = 0
+
+    def get_error_message(self) -> str:
+        if self.state == 0:
+            return "unknown"
+        if self.state == OstreeResult.REMOTE_ADD:
+            return f"error adding remote {self.ostree_error}"
+        if self.state == OstreeResult.GPG_IMPORT:
+            return f"error importing gpg key {self.ostree_error}"
+        if self.state == OstreeResult.PULL:
+            return f"error pulling commit {self.ostree_error}"
+        if self.state == OstreeResult.REMOTE_DELETE:
+            return f"error deleting remote {self.ostree_error}"
+        return None
```

Comment:
The whole state machine is clever, but has the downside that it is de-localized. I think we should probably just have `ostree` return an error code and and do very simple

```
r, msg = ostree()
if r != 0:
   self.signal_error(REMOTE_ADD, "error adding remote: " + msg)
   return 

```

Reply:
I kept some of the ideas but removed the class itself.